core/src/main/java/org/testcontainers/utility/AuditLogger.java

```diff
@@ -25,14 +28,28 @@
     private static final ObjectMapper objectMapper = new ObjectMapper();
     public static final String MDC_PREFIX = AuditLogger.class.getCanonicalName();
 
-    public static void doLog(String action, String image, String containerId, DockerCmd<?> cmd) {
+    public static void doLog(@NotNull String action,
+                             @Nullable String image,
+                             @Nullable String containerId,
+                             @NotNull DockerCmd<?> cmd) {
+
         doLog(action, image, containerId, cmd, null);
     }
 
-    public static void doLog(String action, String image, String containerId, DockerCmd<?> cmd, @Nullable Exception e) {
-        MDC.put(MDC_PREFIX + ".Action", action);
-        MDC.put(MDC_PREFIX + ".Image", image);
-        MDC.put(MDC_PREFIX + ".ContainerId", containerId);
+    public static void doLog(@NotNull String action,
+                             @Nullable String image,
+                             @Nullable String containerId,
+                             @NotNull DockerCmd<?> cmd,
+                             @Nullable Exception e) {
+
+        if (! log.isTraceEnabled()) {
```

Comment:
What if we make this audit thing completely optional? 

i.e. return AuditClient here ( https://github.com/testcontainers/testcontainers-java/blob/88ec3723bf1a47670e1d9ff9f0f260758eaa0fe7/core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java#L97 ) only if some property is set. This way we are safe from any issue related to it, including MDC NPE handling, reflection, etc :)